### PR TITLE
Clarify how password check is done.

### PR DESCRIPTION
**Pull request by ADmad**
I have seen on numerous occasions people trying to debug auth issues think that the SQL query only checking for username is the source of their problems.

### CODE REVIEW

File: en/controllers/components/authentication.rst [hunk 1]
```diff
@@ -164,6 +164,13 @@ the following keys:
 - ``opaque`` A string that must be returned unchanged by clients. Defaults
   to ``md5($config['realm'])``.
 
+.. note::
+    To find the user record, the database is queried only using username.
+    The password check is done in PHP. This is necessary because hashing
+    algorightms like bcrypt (which is used by default) generate a new hash
+    each time, even for same string and you can't just do simple string
```
Comment (dereuromark) [inline en/controllers/components/authentication.rst]: "for the same"

File: en/controllers/components/authentication.rst [hunk 2]
```diff
@@ -164,6 +164,13 @@ the following keys:
 - ``opaque`` A string that must be returned unchanged by clients. Defaults
   to ``md5($config['realm'])``.
 
+.. note::
+    To find the user record, the database is queried only using username.
```
Comment (ravage84) [inline en/controllers/components/authentication.rst]: "using the username" or "using the given username".

File: en/controllers/components/authentication.rst [hunk 3]
```diff
@@ -164,6 +164,13 @@ the following keys:
...
+    The password check is done in PHP. This is necessary because hashing
+    algorithms like bcrypt (which is used by default) generate a new hash
```
Comment (ravage84) [inline en/controllers/components/authentication.rst]: Commas instead of brackets.

File: en/controllers/components/authentication.rst [hunk 4]
```diff
@@ -164,6 +164,13 @@ the following keys:
...
+    each time, even for same string and you can't just do simple string
```
Comment (ravage84) [inline en/controllers/components/authentication.rst]: "for the same string"

File: en/controllers/components/authentication.rst [hunk 5]
```diff
@@ -164,6 +164,13 @@ the following keys:
...
+    comparison in SQL to check if password matches.
```
Comment (ravage84) [inline en/controllers/components/authentication.rst]: "if the password"

**Pull request merged**